go/nbs/cache.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package nbs
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/attic-labs/noms/go/chunks"
+	"github.com/attic-labs/noms/go/d"
+	"github.com/attic-labs/noms/go/hash"
+)
+
+const (
+	defaultCacheMemTableSize uint64 = 1 << 27 // 128MiB
+)
+
+func NewCache() *NomsBlockCache {
+	dir, err := ioutil.TempDir("", "")
+	d.PanicIfError(err)
+	store := NewLocalStore(dir, defaultCacheMemTableSize)
+	d.Chk.NoError(err, "opening put cache in %s", dir)
+	return &NomsBlockCache{store, dir}
+}
+
+// NomsBlockCache holds Chunks, allowing them to be retrieved by hash or enumerated in hash order.
+type NomsBlockCache struct {
+	chunks *NomsBlockStore
+	dbDir  string
+}
+
+// Insert stores c in the cache. If c is successfully added to the cache,
```

Comment:
you're right. Maybe I don't actually need these semantics if I can get rid of all the goroutine machinery